cmd/nirvana/client/client.go

```diff
@@ -0,0 +1,116 @@
+/*
+Copyright 2018 Caicloud Authors
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package client
+
+import (
+	"fmt"
+	"io/ioutil"
+	"os"
+	"path/filepath"
+
+	"github.com/caicloud/nirvana/log"
+	"github.com/caicloud/nirvana/utils/builder"
+	"github.com/caicloud/nirvana/utils/generators/golang"
+	"github.com/caicloud/nirvana/utils/project"
+	"github.com/spf13/cobra"
+	"github.com/spf13/pflag"
+)
+
+func newClientCommand() *cobra.Command {
+	options := &clientOptions{}
+	cmd := &cobra.Command{
+		Use:   "client /path/to/project",
+		Short: "Create client for project",
+		Long:  options.Manuals(),
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := options.Validate(cmd, args); err != nil {
+				log.Fatalln(err)
+			}
+			if err := options.Run(cmd, args); err != nil {
+				log.Fatalln(err)
+			}
+		},
+	}
+	options.Install(cmd.PersistentFlags())
+	return cmd
+}
+
+type clientOptions struct {
+	Output string
+	Rest   string
+}
+
+func (o *clientOptions) Install(flags *pflag.FlagSet) {
+	flags.StringVar(&o.Output, "output", "", "Output directory for generated client")
+	flags.StringVar(&o.Rest, "rest", "github.com/caicloud/nirvana/rest", "Package of rest client")
```

Comment:
what this? rest client template dir?

Reply:
Yes. Sometimes the standard rest package does not satisfy user's requirements. Users can write another one to replace it.